pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -243,9 +267,10 @@ func getAppliedMcRemediations(r *ReconcileComplianceRemediation, rem *compv1alph
 
 	appliedRemediations := make([]*mcfgv1.MachineConfig, 0, len(scanSuiteRemediations.Items))
 	for i := range scanSuiteRemediations.Items {
-		if scanSuiteRemediations.Items[i].Spec.Type != compv1alpha1.McRemediation {
-			// We'll only merge MachineConfigs
-			// TODO: Add a log line with a very high log level
+		if scanSuiteRemediations.Items[i].Spec.Object == nil {
+			continue
+		}
```

Comment:
Sounds like this condition could be rolled into the function (nil is not an MC..)